src/algebra/algebra/spectrum.lean

```diff
@@ -215,33 +214,22 @@ end
 theorem unit_mem_mul_iff_mem_swap_mul {a b : A} {r : Rˣ} :
   ↑r ∈ σ (a * b) ↔ ↑r ∈ σ (b * a) :=
 begin
-  apply not_iff_not.mpr,
-  simp only [mem_resolvent_set_iff, algebra.algebra_map_eq_smul_one],
-  have coe_smul_eq : ↑r • 1 = r • (1 : A), from rfl,
-  rw coe_smul_eq,
-  simp only [is_unit.smul_sub_iff_sub_inv_smul],
-  have right_inv_of_swap : ∀ {x y z : A} (h : (1 - x * y) * z = 1),
-    (1 - y * x) * (1 + y * z * x) = 1, from λ x y z h,
-      calc (1 - y * x) * (1 + y * z * x) = 1 - y * x + y * ((1 - x * y) * z) * x : by noncomm_ring
-      ...                                = 1                                     : by simp [h],
-  have left_inv_of_swap : ∀ {x y z : A} (h : z * (1 - x * y) = 1),
-    (1 + y * z * x) * (1 - y * x) = 1, from λ x y z h,
-      calc (1 + y * z * x) * (1 - y * x) = 1 - y * x + y * (z * (1 - x * y)) * x : by noncomm_ring
-      ...                                = 1                                     : by simp [h],
-  have is_unit_one_sub_mul_of_swap : ∀ {x y : A} (h : is_unit (1 - x * y)),
-    is_unit (1 - y * x), from λ x y h, by
-      { let h₁ := right_inv_of_swap h.unit.val_inv,
-        let h₂ := left_inv_of_swap h.unit.inv_val,
-        exact ⟨⟨1 - y * x, 1 + y * h.unit.inv * x, h₁, h₂⟩, rfl⟩, },
-  have is_unit_one_sub_mul_iff_swap : ∀ {x y : A},
-    is_unit (1 - x * y) ↔ is_unit (1 - y * x), by
-      { intros, split, repeat {apply is_unit_one_sub_mul_of_swap}, },
-  rw [←smul_mul_assoc, ←mul_smul_comm r⁻¹ b a, is_unit_one_sub_mul_iff_swap],
+  have h₁ : ∀ x y : A, is_unit (1 - x * y) → is_unit (1 - y * x),
+  { refine λ x y h, ⟨⟨1 - y * x, 1 + y * h.unit.inv * x, _, _⟩, rfl⟩,
+    calc (1 - y * x) * (1 + y * (is_unit.unit h).inv * x)
+        = (1 - y * x) + y * ((1 - x * y) * h.unit.inv) * x : by noncomm_ring
+    ... = 1 : by simp only [units.inv_eq_coe_inv, is_unit.mul_coe_inv, mul_one, sub_add_cancel],
+    calc (1 + y * (is_unit.unit h).inv * x) * (1 - y * x)
+        = (1 - y * x) + y * (h.unit.inv * (1 - x * y)) * x : by noncomm_ring
+    ... = 1 : by simp only [units.inv_eq_coe_inv, is_unit.coe_inv_mul, mul_one, sub_add_cancel]},
```

Comment:
This might deserve to be a standalone lemmas for any ring, but I can't convince myself that it does. What do you think?

Reply:
Meh, I think it's likely not that useful for general rings, but I've been wrong before. If you want it as a separate result, I would say PR it and then golf this (because it belongs in a different file anyway if we are going to add it), but I don't think it's worth it.